src/views/StandaloneOfficeAddressFiling.vue

```diff
@@ -351,121 +402,88 @@ export default {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.FAS,
+            routingSlipNumber: filing.header.routingSlipNumber,
+            isPriority: filing.header.priority
+          }
+        } else if (filing.header.bcolAccountNumber) {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.BCOL,
+            bcolAccountNumber: filing.header.bcolAccountNumber,
+            datNumber: filing.header.datNumber,
+            folioNumber: filing.header.folioNumber,
+            isPriority: filing.header.priority
+          }
+        } else if (filing.header.waiveFees) {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.NO_FEE
+          }
+        } else {
+          this.staffPaymentData = {
+            option: StaffPaymentOptions.NONE
+          }
+        }
 
-            // load Annual Report fields
-            if (!filing.changeOfAddress) throw new Error('Missing change of address')
-
-            const changeOfAddress = filing.changeOfAddress.offices
-            if (changeOfAddress) {
-              if (changeOfAddress.recordsOffice) {
-                this.addresses = {
-                  registeredOffice: {
-                    deliveryAddress: changeOfAddress.registeredOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.registeredOffice.mailingAddress
-                  },
-                  recordsOffice: {
-                    deliveryAddress: changeOfAddress.recordsOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.recordsOffice.mailingAddress
-                  }
-                }
-                // use existing Priority and Waive Fees flags
-                // apply this flag to BCOMPs only
-                this.updateFilingData('add', FilingCodes.ADDRESS_CHANGE_BC, this.staffPaymentData.isPriority,
-                  (this.staffPaymentData.option === StaffPaymentOptions.NO_FEE))
-              } else {
-                this.addresses = {
-                  registeredOffice: {
-                    deliveryAddress: changeOfAddress.registeredOffice.deliveryAddress,
-                    mailingAddress: changeOfAddress.registeredOffice.mailingAddress
-                  }
-                }
-                // use existing Priority and Waive Fees flags
-                // apply this flag to all OTHER entity types
-                this.updateFilingData('add', FilingCodes.ADDRESS_CHANGE_OT, this.staffPaymentData.isPriority,
-                  (this.staffPaymentData.option === StaffPaymentOptions.NO_FEE))
-              }
-            }
-          } catch (err) {
-            // eslint-disable-next-line no-console
-            console.log(`fetchData() error - ${err.message}, filing = ${filing}`)
-            this.resumeErrorDialog = true
-            throw new Error('Invalid change of address')
+        // do not restore COA Date
+        // since user cannot change it from the UI, it will always be "today"
```

Comment:
OK, this is a bit different from AR and COD.
But anyway, I'm calling it out for a future developer to understand, and in case we eventually allow an "as of" date for COA.